backend/lucene/src/main/java/org/hibernate/search/backend/lucene/scope/model/impl/LuceneScopeModel.java

```diff
@@ -76,26 +76,44 @@ public EventContext getIndexesEventContext() {
 
 	public <T> LuceneScopedIndexObjectComponent<T> getSchemaObjectNodeComponent(String absoluteFieldPath,
 			IndexSchemaFieldNodeComponentRetrievalStrategy<T> componentRetrievalStrategy) {
+		LuceneScopedIndexObjectComponent<T> result = null;
+		String selectedIndexName = null;
+		LuceneIndexSchemaObjectNode selectedObjectNode = null;
+
 		for ( LuceneIndexModel indexModel : indexModels ) {
-			// TODO HSEARCH-2389 check multi-indexes model incompatibility
+			String indexName = indexModel.getIndexName();
+
 			LuceneIndexSchemaObjectNode objectNode = indexModel.getObjectNode( absoluteFieldPath );
+
+			if ( selectedIndexName != null ) {
+				if ( ( selectedObjectNode == null && objectNode != null ) ||
+						( selectedObjectNode != null && objectNode == null ) ) {
+					throw log.conflictingObjectFieldModel( absoluteFieldPath,
```

Comment:
Same problem as for Elasticsearch: you will throw an exception when one of the indexes simply doesn't declare the field, but you shouldn't.

Also, and it's a problem in the case of Lucene, you are completely ignoring two types of incompatibilities:

* the object nodes are stored differently (nested vs. flattened)
* the object nodes have a different set of children